### PR TITLE
just_install_functions update

### PR DESCRIPTION
**Pull request by drewgilliam**
Minor updates to `just_install_functions`
- `conda-python-install` optional input for additional packages
- `pipenv-install` optional input for python activation script
- Activate travis test for mac
- update test for travis & new build pattern

### CODE REVIEW

File: tests/int/test-just_install_functions.bsh [hunk 1]
```diff
@@ -23,8 +23,10 @@ begin_test "cmake-install"
 (
   setup_test
 
+  # redirect mktemp commands to $TESTDIR (does not work on a mac)
+  [ "${VSI_OS}" != "darwin" ] && export TMPDIR="${TESTDIR}"
```
Comment (andyneff) [inline tests/int/test-just_install_functions.bsh:27]: This is inside a test, won't it return false and fail if not darwin?

Make it a normal if statement

Comment (drewgilliam) [reply tests/int/test-just_install_functions.bsh:27]: If that’s true, why did the test succeed in appveyor?

Comment (andyneff) [reply tests/int/test-just_install_functions.bsh:27]: Ah, because `( set -eu; false && false; echo hi; )` apparently always echo's hi, must be one of those corner cases `set -e` is ignored

**Review by andyneff — APPROVED** [review]
